Src/Tests/Temporal.Sdk.Common.Tests/Serialization/TestRawMemoryPayloadConverter.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.IO;
+using System.Linq;
+using Google.Protobuf;
+using Temporal.Api.Common.V1;
+using Temporal.Serialization;
+using Xunit;
+
+namespace Temporal.Sdk.Common.Tests.Serialization
+{
+    public class TestRawMemoryPayloadConverter
+    {
+        [Fact]
+        public void Test_RawMemoryPayloadConverter_ByteString_Roundtrip()
+        {
+            ByteString bs = ByteString.CopyFrom(0, 0, 1);
+            RawMemoryPayloadConverter instance = new();
+            Payloads p = new();
+            Assert.True(instance.TrySerialize(bs, p));
+            Assert.True(instance.TryDeserialize(p, out ByteString actual));
+            Assert.NotNull(actual);
+            Assert.Equal(3, actual.Length);
+            Assert.Equal(bs.ToByteArray(), actual.ToByteArray());
+        }
+
+#if NETCOREAPP3_1_OR_GREATER
+        [Fact]
+        public void Test_RawMemoryPayloadConverter_ReadonlyMemory_Roundtrip()
+        {
+            Random r = new();
+            byte[] buffer = new byte[10];
+            r.NextBytes(buffer);
+            RawMemoryPayloadConverter instance = new();
+            Payloads p = new();
+            Assert.True(instance.TrySerialize(new ReadOnlyMemory<byte>(buffer), p));
+            Assert.True(instance.TryDeserialize(p, out ReadOnlyMemory<byte> actual));
+            Assert.Equal(buffer.Length, actual.Length);
+            Assert.Equal(buffer.ToArray(), actual.ToArray());
+        }
+
+        [Fact]
+        public void Test_RawMemoryPayloadConverter_Memory_Roundtrip()
+        {
+            Random r = new();
+            byte[] buffer = new byte[10];
+            r.NextBytes(buffer);
+            RawMemoryPayloadConverter instance = new();
+            Payloads p = new();
+            Assert.True(instance.TrySerialize(buffer.AsMemory(), p));
+            Assert.True(instance.TryDeserialize(p, out Memory<byte> actual));
+            Assert.Equal(buffer.Length, actual.Length);
+            Assert.Equal(buffer, actual.ToArray());
+        }
+#endif
+
+        [Fact]
+        public void Test_RawMemoryPayloadConverter_MemoryStream_Roundtrip()
+        {
+            Random r = new();
+            byte[] buffer = new byte[10];
+            r.NextBytes(buffer);
+            using MemoryStream ms = new(buffer);
+            RawMemoryPayloadConverter instance = new();
+            Payloads p = new();
+            Assert.True(instance.TrySerialize(ms, p));
+            Assert.True(instance.TryDeserialize(p, out MemoryStream actual));
+            Assert.Equal(buffer.Length, actual.Length);
+            foreach (byte b in buffer)
+            {
+                byte read = (byte)actual.ReadByte();
+                Assert.Equal(b, read);
+            }
+        }
+
+        [Fact]
+        // TODO: Determine whether this should successfully roundtrip
+        public void Test_RawMemoryPayloadConverter_ByteArray_Roundtrip()
```

Comment:
I really like that we have this test case. The Try methods should return false for the same reason as we do not convert enumerable in other converters: not clear whether user meant one array-arg or multiple byte args.
If people have a byte array to encode as raw, they can use [Payload.Raw(..)](url) at the call site. In the future, if there are user requests, we can add more `Payload.Raw(..)` overloads to support other primitive type arrays.